### PR TITLE
Service worker fixes

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Reverts the service worker to stock so it works in older versions of Chrome.
Adds linting so the service worker won't break like this again.
Makes sure json requests aren't cached.

medic/medic#5646
medic/medic#5621
medic/medic#5653

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: grunt/service-worker.tmpl [hunk 1]
```diff
@@ -133,9 +133,7 @@ self.addEventListener('fetch', function(event) {
     // But it fails when running e2e test
     // this has been attributed by the root url serving both json GET request and the cached HTML page
     if (event.request.headers.get('accept').indexOf('application/json') !== -1) {
-      return event.respondWith(async function() {
-        return fetch(event.request);
-      }());
+      return event.respondWith(fetch(event.request));
```
Comment (garethbowen) [inline grunt/service-worker.tmpl]: I _think_ this is equivalent... it seems to work.

Comment (kennsippell) [reply grunt/service-worker.tmpl]: :P lol

**Comment by garethbowen** [comment]
@kennsippell Can I get another review please? This change:

- Improves the service worker template further
- Exposes the server information without needing credentials (potential security risk here?)
- Intercepts pouch's request for the server info and redirects it to the new endpoint. This is essential because browser caching seems to ignore the headers (including cache control headers) and cache the HTML response which breaks Pouch. Now the HTML and JSON responses can be cached and the right one is reliably returned.

File: grunt/service-worker.tmpl [hunk 2]
```diff
@@ -133,9 +133,7 @@ self.addEventListener('fetch', function(event) {
     // But it fails when running e2e test
     // this has been attributed by the root url serving both json GET request and the cached HTML page
     if (event.request.headers.get('accept').indexOf('application/json') !== -1) {
```
Comment (kennsippell) [inline grunt/service-worker.tmpl]: Do we need `tmpl` at all anymore?

**Comment by kennsippell** [comment]
Also: since you're coincidentally fixing https://github.com/medic/medic/pull/5629 in this PR -- does it make sense to just merge the tests from that PR into this change? Close that PR and get the tests passing and make sure there is test coverage for this change?

**Comment by garethbowen** [comment]
@kennsippell Please have another look. I've merged the other PR in and removed the service worker tmpl.

File: api/src/routing.js [hunk 3]
```diff
@@ -167,12 +167,12 @@ app.use(compression({
 //       https://github.com/medic/medic/issues/4089
 
 app.get('/', function(req, res) {
-  if (req.headers.accept === 'application/json') {
-    // couchdb request - let it go
-    proxy.web(req, res);
-  } else {
-    res.sendFile(path.join(extractedResourceDirectory, 'templates/inbox.html'));
-  }
+  res.sendFile(path.join(extractedResourceDirectory, 'templates/inbox.html'));
```
Comment (kennsippell) [inline api/src/routing.js]: Actually on second thought. When app is running vPrev and we deploy v3.5 we temporarily have v3.5's api serving the requests from vPrev. That code won't know to request to `/dbinfo` so it isn't safe to remove the `if (req.headers.accept === 'application/json')` block here ...

Comment (kennsippell) [reply api/src/routing.js]: Nick was eager to test this today - so I pushed that commit into this PR. Sounds like the upgrade is now unblocked.

Comment (garethbowen) [reply api/src/routing.js]: Looks good to me. I think this is ready for AT.

**Comment by kennsippell** [comment]
@garethbowen I think this is good to merge. Rebasing to prompt test re-run.

**Comment by kennsippell** [comment]
Merged after conversation in today's product meeting despite some unresolved issues with AT during upgrade.

**Comment by garethbowen** [comment]
@kennsippell Does that mean we can close #5646 and #5621 ?

**Comment by kennsippell** [comment]
@garethbowen I do believe so. But I'll let AT close them when they are satisfied @newtewt. Leaving #5653 open to track the update issue.

**Pull request merged**